pallets/dapp-staking-v3/src/lib.rs

```diff
@@ -926,6 +1197,336 @@ pub mod pallet {
 
             Ok(())
         }
+
+        /// Claims some staker rewards, if user has any.
+        /// In the case of a successfull call, at least one era will be claimed, with the possibility of multiple claims happening
+        /// if appropriate entries exist in account's ledger.
+        #[pallet::call_index(11)]
+        #[pallet::weight(Weight::zero())]
+        pub fn claim_staker_rewards(origin: OriginFor<T>) -> DispatchResult {
+            Self::ensure_pallet_enabled()?;
+            let account = ensure_signed(origin)?;
+
+            let mut ledger = Ledger::<T>::get(&account);
+            let staked_period = ledger
+                .staked_period()
+                .ok_or(Error::<T>::NoClaimableRewards)?;
+
+            // Check if the rewards have expired
+            let protocol_state = ActiveProtocolState::<T>::get();
+            ensure!(
+                staked_period >= Self::oldest_claimable_period(protocol_state.period_number()),
```

Comment:
If `staked period` is < of `oldest claimable period` no fund can be claimed as this ensure! will prevent from it.
In my undestanding staker can still claim the rewards that are between` oldest claimable period` and `now`

Reply:
Had a second look and I got it. Actually `.staked_period()` return the last period eligible for rewards.
I was afraid that, if an user stake for let's say 1 year without claiming, even his from  oldest claimable period` to `now` staking era will now be claimable. But they actually will